### PR TITLE
fix #6153 add genesis time dependence simulations

### LINKED ISSUE #6153: time-depenence genesis simulations
Add the time-dependence fields and the ITDP flag to enable that mode.

### PR DESCRIPTION
**Pull request by moellep**

### CODE REVIEW

File: sirepo/template/genesis.py [hunk 1]
```diff
@@ -232,6 +244,16 @@ def _get_col(col_key):
...
 
 
+def validate_file(file_type, path):
+    if file_type == "io-partfile":
+        if _is_text_file(path):
+            return "The PARTFILE should be a binary file. Use the DISTFILE to import a text file."
+    elif file_type == "io-distfile":
+        if not _is_text_file(path):
+            return "The DISTFILE should be a text file with columns: X PX Y PY T P."
+    return None
```
Comment (robnagler) [inline sirepo/template/genesis.py:254]: Do we assume that if it doesn't match a known type, it's correct?

File: sirepo/template/genesis.py [hunk 2]
```diff
@@ -327,6 +358,15 @@ def _get_frame_counts(run_dir):
     return res
 
 
+def _is_text_file(path):
+    try:
+        pkio.read_text(path)
```
Comment (robnagler) [inline sirepo/template/genesis.py:363]: If the files are large, this is expensive (reads the whole file). If the file is not large, probably fine.

we may want to consider binaryornot, which is based on Perl's algorithm.

File: sirepo/template/genesis.py [hunk 3]
```diff
@@ -327,6 +358,15 @@ def _get_frame_counts(run_dir):
...
+        return True
+    except UnicodeDecodeError:
+        pass
```
Comment (robnagler) [inline sirepo/template/genesis.py:366]: `return False` here so less code.

File: sirepo/template/genesis.py [hunk 4]
```diff
@@ -367,14 +407,13 @@ def _parse_namelist(data, text):
...
-    # TODO(pjm): remove this if scanning or time dependence is implemented in the UI
+    # TODO(pjm): remove this if scanning is implemented in the UI
     dm.scan.iscan = "0"
-    dm.timeDependence.itdp = "0"
     return data
 
 
 def _z_title_at_frame(frame_args, nth):
-    _, s = _get_lattice_and_slice_data(frame_args.run_dir)
+    _, s = _get_lattice_and_slice_data(frame_args.run_dir, 0)
```
Comment (robnagler) [inline sirepo/template/genesis.py:416]: I tend to prefer: 

```py
s = _get_lattice_and_slice_data(frame_args.run_dir, 0)[1]
```

**Review by robnagler — CHANGES_REQUESTED** [review]
Looks fine to me, just some style comments.

File: sirepo/template/genesis.py (hunk 1 above)
Comment (moellep) [reply sirepo/template/genesis.py:254]: That's correct - I added input file checking for 2 common cases, which caused confusion in the omega webinar prep.

File: sirepo/template/genesis.py (hunk 2 above)
Comment (moellep) [reply sirepo/template/genesis.py:363]: Yes, I looked at binaryornot but we do not currently have it as part of the base install, and genesis input files are not overly large.

**Pull request merged**